airvantage.go

```diff
@@ -101,8 +105,69 @@ type apiError struct {
 func (av *AirVantage) parseResponse(resp *http.Response, respStruct interface{}) error {
 	defer resp.Body.Close()
 
+	if err := av.parseError(resp); err != nil {
+		return err
+	}
+
+	if respStruct == nil {
+		return nil
+	}
```

Comment:
I find the `if <INSTRUCTION>; <TEST> {}` unusual 😅 
If you want to save lines:
```

err := av.parseError(resp)
if err != nil || respStruct == nil {
   return err
}
```

Reply:
In fact, the `respStruct == nil` doesn't return an error. I guess this test is more a sanity check. it is a bit weird (why would I try to parse a response with provide a result object pointer?) but it's sort of make sense (if we do this but get an HTTP error, we might want to catch it).